dm/syncer/syncer.go

```diff
@@ -2298,6 +2298,8 @@ func (s *Syncer) handleQueryEvent(ev *replication.QueryEvent, ec eventContext, o
 		if err2 != nil {
 			return err2
 		}
+		// adjust 'Collation' base on upstream
+		s.adjustTableCollation(ddlInfo)
```

Comment:
how about batch add collation option to those ddl sql after this loop?

Reply:
yes, if it can merge collation search into one statement, it may improve performance. i think i will try later.